optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/common/inliner/AbstractScoreInliner.java

```diff
@@ -0,0 +1,212 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.common.inliner;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.score.buildin.bendable.BendableScore;
+import org.optaplanner.core.api.score.buildin.bendablebigdecimal.BendableBigDecimalScore;
+import org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScore;
+import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
+import org.optaplanner.core.api.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScore;
+import org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScore;
+import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
+import org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScore;
+import org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
+import org.optaplanner.core.api.score.buildin.simple.SimpleScore;
+import org.optaplanner.core.api.score.buildin.simplebigdecimal.SimpleBigDecimalScore;
+import org.optaplanner.core.api.score.buildin.simplelong.SimpleLongScore;
+import org.optaplanner.core.api.score.constraint.ConstraintMatch;
+import org.optaplanner.core.api.score.constraint.ConstraintMatchTotal;
+import org.optaplanner.core.api.score.constraint.Indictment;
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.impl.score.buildin.BendableBigDecimalScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.BendableLongScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.BendableScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.HardMediumSoftBigDecimalScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.HardMediumSoftLongScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.HardMediumSoftScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.HardSoftBigDecimalScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.HardSoftLongScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.HardSoftScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.SimpleBigDecimalScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.SimpleLongScoreDefinition;
+import org.optaplanner.core.impl.score.buildin.SimpleScoreDefinition;
+import org.optaplanner.core.impl.score.constraint.DefaultConstraintMatchTotal;
+import org.optaplanner.core.impl.score.constraint.DefaultIndictment;
+import org.optaplanner.core.impl.score.definition.ScoreDefinition;
+
+public abstract class AbstractScoreInliner<Score_ extends Score<Score_>> {
```

Comment:
Shouldn't this also implement ScoreInliner?

Reply:
There is none. :-)